src/libs/actions/PersonalDetails.ts

```diff
@@ -1,97 +1,90 @@
+        currentUserEmail = val?.email ?? '';
+        currentUserAccountID = val?.accountID ?? -1;
     },
 });
 
-let allPersonalDetails;
+let allPersonalDetails: OnyxEntry<Record<string, PersonalDetails>> = null;
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
     callback: (val) => (allPersonalDetails = val),
 });
 
-let privatePersonalDetails;
+let privatePersonalDetails: OnyxEntry<PrivatePersonalDetails> = null;
 Onyx.connect({
     key: ONYXKEYS.PRIVATE_PERSONAL_DETAILS,
     callback: (val) => (privatePersonalDetails = val),
 });
 
 /**
  * Returns the displayName for a user
- *
- * @param {String} login
- * @param {Object} [personalDetail]
- * @returns {String}
  */
-function getDisplayName(login, personalDetail) {
+function getDisplayName(login: string, personalDetail: Pick<PersonalDetails, 'firstName' | 'lastName'> | null): string {
     // If we have a number like +15857527441@expensify.sms then let's remove @expensify.sms and format it
     // so that the option looks cleaner in our UI.
     const userLogin = LocalePhoneNumber.formatPhoneNumber(login);
-    const userDetails = personalDetail || lodashGet(allPersonalDetails, login);
+    const userDetails = personalDetail ?? allPersonalDetails?.[login];
 
     if (!userDetails) {
         return userLogin;
     }
 
-    const firstName = userDetails.firstName || '';
-    const lastName = userDetails.lastName || '';
+    const firstName = userDetails.firstName ?? '';
+    const lastName = userDetails.lastName ?? '';
     const fullName = `${firstName} ${lastName}`.trim();
 
     return fullName || userLogin;
 }
 
 /**
- * @param {String} userAccountIDOrLogin
- * @param {String} [defaultDisplayName] display name to use if user details don't exist in Onyx or if
+ * @param [defaultDisplayName] display name to use if user details don't exist in Onyx or if
  *                                      found details don't include the user's displayName or login
- * @returns {String}
  */
-function getDisplayNameForTypingIndicator(userAccountIDOrLogin, defaultDisplayName = '') {
+function getDisplayNameForTypingIndicator(userAccountIDOrLogin: string, defaultDisplayName = ''): string {
     // Try to convert to a number, which means we have an accountID
     const accountID = Number(userAccountIDOrLogin);
 
     // If the user is typing on OldDot, userAccountIDOrLogin will be a string (the user's login),
     // so Number(string) is NaN. Search for personalDetails by login to get the display name.
-    if (_.isNaN(accountID)) {
-        const detailsByLogin = _.findWhere(allPersonalDetails, {login: userAccountIDOrLogin}) || {};
-        return detailsByLogin.displayName || userAccountIDOrLogin;
+    if (Number.isNaN(accountID)) {
+        const detailsByLogin = Object.entries(allPersonalDetails ?? {}).find(([, value]) => value?.login === userAccountIDOrLogin)?.[1];
+        return detailsByLogin?.displayName ?? userAccountIDOrLogin;
```

Comment:
This will cause regression when `detailsByLogin.displayName` is empty string and `userAccountIDOrLogin`exists